edgexfoundry/app-rfid-llrp-inventory
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.4.1 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.4.1)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use log package for logging
- Use library function for autostart processing

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-rfid-llrp-inventory/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)  [app-rfid-llrp-inventory test suite](https://github.com/canonical/edgex-snap-testing/tree/main/test/suites/app-rfid-llrp-inventory) passed with `FULL_CONFIG_TEST=true`
  <link to docs PR>

## Testing Instructions

1. Install and validate that all config files have been copied in right place:
```
tree /var/snap/edgex-app-rfid-llrp-inventory/current/config
.
/var/snap/edgex-app-rfid-llrp-inventory/current/config
└── app-rfid-llrp-inventory
    └── res
        └── configuration.toml

2 directories, 1 file
```
2. Setup and start:
```
sudo snap set edgex-app-rfid-llrp-inventory app-options=true
sudo snap set edgex-app-rfid-llrp-inventory config.writable-loglevel=DEBUG
sudo snap set edgex-app-rfid-llrp-inventory config.service-startupmsg="#test start up message#"
sudo snap start edgex-app-rfid-llrp-inventory
```

3. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-app-rfid-llrp-inventory/current/config/app-rfid-llrp-inventory/res/app-rfid-llrp-inventory.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
4. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs edgex-app-rfid-llrp-inventory

msg="#test start up message#"
level=DEBUG msg="xxx"
```
5. Validate autostart via snap option works:
```
$ sudo snap set edgex-app-rfid-llrp-inventory autostart=true
$ snap services edgex-app-rfid-llrp-inventory
Service                                                Startup  Current  Notes
edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory  enabled  active   -

$ sudo snap set edgex-app-rfid-llrp-inventory autostart=false
$ snap services edgex-app-rfid-llrp-inventory
Service                                                Startup   Current   Notes
edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory  disabled  inactive  -
```

## New Dependency Instructions (If applicable)
